ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/BraveTranslateScriptHandler.swift

```diff
@@ -0,0 +1,665 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BraveCore
+import Combine
+import Foundation
+import NaturalLanguage
+import Preferences
+import Shared
+import SwiftUI
+import Translation
+import WebKit
+import os.log
+
+protocol BraveTranslateScriptHandlerDelegate: NSObject {
+  func updateTranslateURLBar(tab: Tab?, state: TranslateURLBarButton.TranslateState)
+  func showTranslateOnboarding(tab: Tab?, completion: @escaping (_ translateEnabled: Bool) -> Void)
+  func presentToast(tab: Tab?, languageInfo: BraveTranslateLanguageInfo)
+}
+
+class BraveTranslateScriptHandler: NSObject, TabContentScript {
+  private weak var tab: Tab?
```

Comment:
We need to refactor this script handler to not hold onto a tab for the upcoming CWVWebView usage. The `tab` will only be passed in the `didReceiveScriptMessage` delegate method

Reply:
To clarify, for this branch you can pass in a Tab and hold onto it, but we shouldn't use it anywhere outside of the didReceiveScriptMessage content controller delegate method (pass it to other methods) to ensure we can easily use the same script handler in the cwv-webview branch